components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt

```diff
@@ -339,6 +349,40 @@ class AddonsManagerAdapter(
             }
         }
     }
+
+    /**
+     * Refresh only the portion of the list that contains the [addon].
+     * @property addon The add-on to be update.
+     */
+    fun refresh(addon: Addon){
+        addonsMap[addon.id] = addon
+        submitList(createListWithSections(addonsMap.values.toList()))
+    }
+
+    /**
+     * Refresh only the portion of the list that changes between the currentList and the new [addons].
+     * @property addons A list of add-on to replace the actual list.
+     */
+    fun refresh(addons: List<Addon>){
+        addonsMap = addons.associateBy({ it.id }, { it }).toMutableMap()
+        submitList(createListWithSections(addons))
```

Comment:
As the list that we show to the users has multiple sections (**Recommended**, **installed**, **disabled**, **unsupported**), we need to re build a new list to diff it with the current list, because changes to the ad-ons can cause changes to the the sections, for example if we have 2 add-on and all of them are **Recommended**, if we install one the new list will have two new sections **installed**, and  **Recommended**. Now this is not an issue as the list is quite small, but when we have many +200 it could be an issue.